sdk/commands/sensor_test.ts

```diff
@@ -107,74 +103,118 @@ export class SMXPanelTestData {
       return;
     }
 
+    // Assuming the sig bits are correct, we can confirm here that we have proper data
     this.have_data_from_panel = true;
 
     /**
      * These bits are true if that sensor's most recent reading is invalid.
      * A sensors reading could be considered invalid if the sensor has been turned
      * off in the config tool.
      */
-    this.bad_sensor_input = {
-      up: data.sig_bad.bad_sensor_0,
-      right: data.sig_bad.bad_sensor_1,
-      down: data.sig_bad.bad_sensor_2,
-      left: data.sig_bad.bad_sensor_3,
-    };
+    this.bad_sensor_input = [
+      data.sig_bad.bad_sensor_0,
+      data.sig_bad.bad_sensor_1,
+      data.sig_bad.bad_sensor_2,
+      data.sig_bad.bad_sensor_3,
+    ];
 
     // This is what the dipswitch is set to for this panel
     this.dip_switch_value = data.dips.dip;
 
     // These are true if the sensor has the incorrect jumper set
-    this.bad_jumper = {
-      up: !!data.dips.bad_sensor_dip_0,
-      right: !!data.dips.bad_sensor_dip_1,
-      down: !!data.dips.bad_sensor_dip_2,
-      left: !!data.dips.bad_sensor_dip_3,
-    };
+    this.bad_jumper = [
+      !!data.dips.bad_sensor_dip_0,
+      !!data.dips.bad_sensor_dip_1,
+      !!data.dips.bad_sensor_dip_2,
+      !!data.dips.bad_sensor_dip_3,
+    ];
 
     /**
      * These are 16-bit signed integers for the sensor values.
      * These are signed as they can be negative, but I imagine them going
      * negative is just kind of noise from the hardware.
      */
-    this.sensor_level = {
-      left: data.sensors[0],
-      right: data.sensors[1],
-      up: data.sensors[2],
-      down: data.sensors[3],
-    };
+    this.sensor_level = data.sensors.map((value) => this.clamp_sensor_value(value, mode));
+  }
+
+  private clamp_sensor_value(value: number, mode: SensorTestMode) {
+    if (mode === SensorTestMode.Noise) {
+      /**
+       * In Noise mode, we receive standard deviation values squared.
+       * Display the square root, since the panels don't do this for us.
+       * This makes the number different than the configured value
+       * (square it to convert back), but without this we display a bunch
+       * of four and five digit numbers that are too hard to read.
+       *
+       * TODO: Do we want to round this value or just display decimal values?
+       */
+      return Math.sqrt(value);
```

Comment:
We can also resolve this by just displaying the squared values as they come across the wire in a more abbreviated form e.g. `12k` instead of `12345`. I'm not even sure what the value in the noise test mode is, so we can wait to decide later.